src/server/string_family.cc

```diff
@@ -557,14 +558,18 @@ void StringFamily::Set(CmdArgList args, ConnectionContext* cntx) {
       sparams.flags |= SetCmd::SET_KEEP_EXPIRE;
     } else if (cur_arg == "GET") {
       sparams.flags |= SetCmd::SET_GET;
+      sparams.prev_val = &prev_val;
     } else {
       return builder->SendError(kSyntaxErr);
     }
   }
 
   const auto result{SetGeneric(cntx, sparams, key, value)};
-
-  if (result == OpStatus::OK) {
+  if (result == OpStatus::OK && sparams.flags & SetCmd::SET_GET && prev_val) {
```

Comment:
1. I prefer hierarchical IFs where you first check `if (result) {`, and inside all other checks related to the branch of result is ok.
2. when using `&&` and `&` , please add parentheses for readability around `&` condition?
3. I did not quite understand how it happens that `sparams.flags & SetCmd::SET_GET` but prev_val is  null.